library/authenticators.c

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright (c) 2019-2020 NetFoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "authenticators.h"
+
+typedef struct authenticator_ctx_s {
+    ziti_context ztx;
+    void* ctx;
+    char* csr_pem;
+    char* authenticator_id;
+} authenticator_ctx;
+
+static void send_error_event(authenticator_ctx* authenticator_ctx, int err){
+    ziti_context ztx = authenticator_ctx->ztx;
+
+    ziti_event_t ev = {
+            .type = ZitiAuthenticatorEvent,
+            .event.authenticator = {
+                    .ctx = authenticator_ctx->ctx,
+                    .error = err,
+                    .is_verified = false,
+                    .authenticator_id = authenticator_ctx->authenticator_id,
+            }
+    };
+
+    ziti_send_event(ztx, &ev);
+}
+
+const char* CAN_NOT_UPDATE_AUTHENTICATOR = "CAN_NOT_UPDATE_AUTHENTICATOR";
```

Comment:
Should CAN_NOT be one word: CANNOT?

Reply:
Quick googling shows that when stating something is not possible both forms are correct. 

I am inclined to take no action.